src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -154,13 +58,13 @@ public class TagHelperDescriptor
         /// <remarks>
         /// <c>*</c> at the end of an attribute name acts as a prefix match.
         /// </remarks>
-        public IReadOnlyList<string> RequiredAttributes { get; }
+        public IEnumerable<string> RequiredAttributes { get; [param: NotNull] set; } = Enumerable.Empty<string>();
 
         /// <summary>
         /// HTML element tag names targeted by tag helpers allowed as children.
         /// </summary>
         /// <remarks><c>null</c> indicates all tag helper children are allowed.</remarks>
-        public IReadOnlyList<string> AllowedChildren { get; }
+        public IEnumerable<string> AllowedChildren { get; set; }
```

Comment:
Ah, I understand. I checked and It looks like the instances are used only to create the instance of the `TagHelperDescriptor`. So, we're okay.